src/HSE/All.hs

```diff
@@ -229,6 +231,24 @@ ghcFailOpParseModuleEx ppstr file str (loc, err) = do
                ErrUtils.pprLocErrMsg (ErrUtils.mkPlainErrMsg baseDynFlags loc err)
    return $ Left $ ParseError sl msg pe
 
+-- | Produce a pair of lists from a 'ParseFlags' value representing
+--   language extensions to explicitly enable/disable.
+ghcExtensionsFromParseFlags :: ParseFlags
+                             -> ([GHC.Extension], [GHC.Extension])
+ghcExtensionsFromParseFlags ParseFlags {hseFlags=ParseMode {extensions=exts}}=
+    foldl' classify ([], []) exts
```

Comment:
See partitionEither for an easier way to do this.

Reply:
Ok.